### PR TITLE
feat: [PL-23871]: Pagination in switch account

### PR DESCRIPTION
**Pull request by avinashmadhwani02**
#### Screenshots

Screenshot 2022-07-21 at 9 27 29 PM

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by avinashmadhwani02** [comment]
dockerize

**Comment by avinashmadhwani02** [comment]
dockerize

File: src/modules/10-common/modals/SwitchAccount/SwitchAccount.tsx [hunk 1]
```diff
@@ -206,19 +219,34 @@ const SwitchAccount: React.FC<SwitchAccountProps> = ({ searchString = '', mock }
...
-    [accounts]
+    [accounts, currentUserInfo]
   )
   return (
     <>
       <ReAuthenticationNote accounts={accounts} accountId={accountId} />
       <Container padding={{ left: 'large', right: 'large' }} className={css.container}>
-        {loading || settingDefault ? <PageSpinner /> : null}
+        {loading || userInfoLoading || settingDefault ? (
+          <Container className={css.spinner}>
```
Comment (abhinavrastogi-harness) [inline src/modules/10-common/modals/SwitchAccount/SwitchAccount.tsx:229]: you can remove the container the `spinner` css. PageSpinner should already be taking full screen. 
if it's not, we can check the layout changes

File: src/modules/10-common/modals/SwitchAccount/SwitchAccount.module.scss [hunk 2]
```diff
@@ -25,3 +26,26 @@
     max-height: 500px !important;
   }
 }
+
+.spinner {
```
Comment (abhinavrastogi-harness) [inline src/modules/10-common/modals/SwitchAccount/SwitchAccount.module.scss:30]: this class should not be needed

File: src/modules/10-common/modals/SwitchAccount/SwitchAccount.tsx [hunk 3]
```diff
@@ -15,26 +15,29 @@ import {
...
+import type { Account } from 'services/portal'
 import { PageSpinner } from '@common/components'
 import type { AccountPathProps } from '@common/interfaces/RouteInterfaces'
 import { useStrings } from 'framework/strings'
 import routes from '@common/RouteDefinitions'
 import type { UseGetMockData } from '@common/utils/testUtils'
 import { getLoginPageURL } from 'framework/utils/SessionUtils'
 import SecureStorage from 'framework/utils/SecureStorage'
+import { useAppStore } from 'framework/AppStore/AppStoreContext'
+import { useGetCurrentUserInfo } from 'services/cd-ng'
```
Comment (abhinavrastogi-harness) [inline src/modules/10-common/modals/SwitchAccount/SwitchAccount.tsx:40]: currentUserInfo is already available in appstore context. why do need to make a new network call here?

File: src/modules/10-common/modals/SwitchAccount/SwitchAccount.tsx [hunk 4]
```diff
@@ -60,27 +63,51 @@ const RenderColumnAccountEdition: Renderer<CellProps<Account>> = ({ row }) => {
...
+    queryParams: { accountIdentifier: accountId },
+    lazy: true
   })
   const { mutate: setDefaultAccount, loading: settingDefault } = useSetDefaultAccountForCurrentUser({ accountId })
   const { mutate: switchAccount, loading: switchAccountLoading } = useRestrictedSwitchAccount({
     // requestOptions: { headers: { 'content-type': 'application/json' } }
   })
 
+  useEffect(() => {
+    if (userInfo?.data?.defaultAccountId !== currentUserInfo.defaultAccountId) {
```
Comment (abhinavrastogi-harness) [inline src/modules/10-common/modals/SwitchAccount/SwitchAccount.tsx:104]: why is this needed? let's discuss on a call

**Review by abhinavrastogi-harness — CHANGES_REQUESTED** [review]

**Review by abhinavrastogi-harness — APPROVED** [review]

**Review by amitsh14 — APPROVED** [review]

**Pull request merged**